packages/react/src/ReactLazy.js

```diff
@@ -48,9 +48,13 @@ export type LazyComponent<T, P> = {
   _init: (payload: P) => T,
 };
 
-function lazyInitializer<T>(payload: Payload<T>): T {
+const lazyInitializer = (ctor: Function) => <T>(payload: Payload<T>): T => {
```

Comment:
Assumption is that init function is called to initialize the lazy retrieval - and that this will only occur on initial render (or remount).